### ISSUE TITLE
mDec: 0 and Postback error

### ISSUE THREAD

**Issue by tiborjohnson** [BobKnothe/autoNumeric#67]
When I set mDec: 0, after postback trailing 0's are truncated from the text.  For example if i enter in 100,000, after postback the entry is changed to 100.  

I have noticed this is an issue when there is only one ',' in the value.  any other length is accepted.
Test code below:

<%--html xmlns="http://www.w3.org/1999/xhtml">
head runat="server">
    script src="../scripts/jquery-1.11.0.min.js" type="text/javascript"></script>
    script src="../scripts/autoNumeric.js" type="text/javascript"></script>
/head>
body>
    form id="form1" runat="server">
    div>
        asp:TextBox ID="TextBox1" runat="server"></asp:TextBox>

        asp:Button ID="Button1" runat="server" Text="Button" />
    /div>
    /form>
    script>
        jQuery(function ($) {
            $("#<%=TextBox1.ClientID %>").autoNumeric('init', { vMax: '999999', mDec: 0 });

                });
    /script>
/body>
/html>--%>

**Comment by glt101** [comment]
Hi Bob,

Me too. If I use mDec: '0', then zeros after the thousand separator are truncated on postback.
This happens regardless of whether I use mDec:'0' or use the vMin and vMax set to integers.

Any resolution?

Cheers,
Geoff

**Comment by BobKnothe** [comment]
Geoff,

Happy New Year.

autoNumeric by default, formats default values of the input. On NON-ASP 
pages during re-load autoNumeric compares the default value to the 
updated value and determines if formatting is required. however ASP.NET 
is unique (most variations) because on postback (page re-load) the input 
tag is re-rendered and it over writes the default value with the last 
value entered so autoNumeric "assumes" the new value needs to be 
formatted leading to possible errors.

So there are two possible solutions:

 1. If there are no default value(s) change the "aForm" option from
    "true" to "false"
 2. If there are default values and they are formatted on page load use
    the option "anDefault" which should be set to the same value as the
    input default value
      * example input default value="1234.56"
      * then the option {anDeFault: "1234.56"} should be set to the same
        value as the default value. Please not this can be set as an
        option of HTML 5 data attribute.

Hope this helps.

Bob

**Comment by glt101** [comment]
Hi Bob,

A Happy New Year to you too and thanks for the reply. I see now that you have this on your GitHub page, so this is, unsurprisingly, something of an FAQ. Sorry I missed that!

However, I still can't get the fix to work. Even if I add the anDefault on postback:

`TextBoxDesignFees.Attributes.Add("data-an-default", TextBoxDesignFees.Text);`

the value in the textBox gets truncated.
See my attached example page. 

Cheers,
Geoff
[t1.zip](https://github.com/BobKnothe/autoNumeric/files/83007/t1.zip)

**Comment by BobKnothe** [comment]
I am no ASP expert but the two values do not equal each other.



protected void Page_Load(object sender, EventArgs e)
{
     //Pretend a value was read from a data source on first page load:
     if (!Page.IsPostBack)
     {
         TextBoxDesignFees.Text = "_2000000.00_";
         TextBoxDesignFees.Attributes.Add("data-an-default", "_20000.00_");
     }
     else
     {
         //On postback, add the fix for zeros truncation to the Input tag
         TextBoxDesignFees.Attributes.Add("data-an-default", 
TextBoxDesignFees.Text);
     }
}

**Comment by glt101** [comment]
Ok. True. Correcting that doesn't change the behavior though.
[t1.zip](https://github.com/BobKnothe/autoNumeric/files/83115/t1.zip)

Its looking more promising for me if I use aForm: false. At least the truncation problem goes away if I use that. Now I just need to format the initial value using .Net I guess.

Cheers,
Geoff

**Comment by BobKnothe** [comment]
Geoff,

Rather then add "anDefault" as an HTML5 data attribute have you tried 
adding as an option when "init" the input 
$(selector).autoNumeric('init', {anDefault: "200000.00"});?

Bob

**Comment by glt101** [comment]
Heh. I was doing something very similar.
I used the 'set' method to apply the formatting manually once the page loaded.
I was able to construct the javascript for that in the C# and emit the script at the end of the page. This seems to correct the truncation problem after it has happened.
The attached version seems to work ok now.

Thanks for the help :)

Cheers,
Geoff

[t1.zip](https://github.com/BobKnothe/autoNumeric/files/83238/t1.zip)
